src/core/public/overlays/banners/banners_service.tsx

```diff
@@ -0,0 +1,148 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import React from 'react';
+import { BehaviorSubject, Observable } from 'rxjs';
+import { map } from 'rxjs/operators';
+
+import { PriorityMap } from './priority_map';
+import { BannersList } from './banners_list';
+import { UiSettingsClientContract } from '../../ui_settings';
+import { I18nStart } from '../../i18n';
+import { UserBannerService } from './user_banner_service';
+
+/**
+ * A function that will unmount the banner from the element.
+ * @public
+ */
+export type OverlayBannerUnmount = () => void;
+
+/**
+ * A function that will mount the banner inside the provided element.
+ * @param element an element to render into
+ * @returns a {@link OverlayBannerUnmount}
+ * @public
+ */
+export type OverlayBannerMount = (element: HTMLElement) => OverlayBannerUnmount;
+
+/** @public */
+export interface OverlayBannersStart {
+  /**
+   * Add a new banner
+   *
+   * @param mount {@link OverlayBannerMount}
+   * @param priority optional priority order to display this banner. Higher priority values are shown first.
+   * @returns a unique identifier for the given banner to be used with {@link OverlayBannersStart.remove} and
+   *          {@link OverlayBannersStart.replace}
+   */
+  add(mount: OverlayBannerMount, priority?: number): string;
+
+  /**
+   * Remove a banner
+   *
+   * @param id the unique identifier for the banner returned by {@link OverlayBannersStart.add}
+   * @returns if the banner was found or not
+   */
+  remove(id: string): boolean;
+
+  /**
+   * Replace a banner in place
+   *
+   * @param id the unique identifier for the banner returned by {@link OverlayBannersStart.add}
+   * @param mount {@link OverlayBannerMount}
+   * @param priority optional priority order to display this banner. Higher priority values are shown first.
+   * @returns a new identifier for the given banner to be used with {@link OverlayBannersStart.remove} and
+   *          {@link OverlayBannersStart.replace}
+   */
+  replace(id: string | undefined, mount: OverlayBannerMount, priority?: number): string;
```

Comment:
I was surprised that `replace(undefined, () => ..)` adds a new banner in the list. can we force `add` usage for such cases? Legacy platform doesn't provide this API and `id` is required parameter https://github.com/elastic/kibana/blob/5716bfde16b33d457aeac4c0a5559de22fd0c580/src/legacy/ui/public/notify/banners/banners.tsx#L75